lib/post_creator.rb

```diff
@@ -107,8 +107,8 @@ def valid?
       topic_creator = TopicCreator.new(@user, guardian, @opts)
       return false unless skip_validations? || validate_child(topic_creator)
     else
-      @topic = Topic.find_by(id: @opts[:topic_id])
-      if (@topic.blank? || !guardian.can_create?(Post, @topic))
+      @topic = Topic.with_deleted.find_by(id: @opts[:topic_id])
+      if (@topic.blank? || !guardian.can_create?(Post, @topic)) && !skip_validations?
```

Comment:
I'm abit worried here because of the comment in

https://github.com/discourse/discourse/blob/80e348d226583d7dbbb48a914ced96d9fef60490/lib/guardian/topic_guardian.rb#L22

I would rather we pass in an additional option to allow a post to be created in a deleted topic rather than making the condition dependent on skip_validations